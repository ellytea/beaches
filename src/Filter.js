import React, { Component } from 'react';


class Filter extends Component {
  constructor(){
    super();
  }

  render(){
    return(
      <div className='filter-dropdown'>
        <select onChange={this.props.filterByCounty}
          className='filter'
          placeholder='Filter by County'>
        <option className='filter-names' value='all-counties'>All Counties</option>
        {
          this.props.beaches.map((beach) => {
            return <option className='filter-names' value={beach.county}>{beach.county}</option>
          })
        }
        </select>
      </div>

    )
  }
}

export default Filter;
