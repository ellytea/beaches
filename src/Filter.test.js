import React from 'react';
import Filter from './Filter.js';
import { mount, shallow } from 'enzyme';

const filterByCountyMock = jest.fn()

const beachCounties = [
  {
      "name": "San Luis Obispo County",
      "beaches": [ "MontaÃ±a de Oro State Beach", "Cayucos State Beach", "Pismo State Beach", "William Randolph Hearst Memorial State Beach" ]
  }];

describe('Filter', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Filter filterByCounty={filterByCountyMock}
              beachCounties={beachCounties} />
      )
  })

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should invoke filterByCounty function', () => {
    wrapper.find('.filter').simulate('change', { target: {value: "San Luis Obispo County"} })
    expect(filterByCountyMock).toBeCalled();
  })

})