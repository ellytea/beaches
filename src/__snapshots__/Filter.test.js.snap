// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match the snapshot with all data passed in 1`] = `
<div
  className="filter-dropdown"
>
  <select
    className="filter"
    onChange={[MockFunction]}
    placeholder="Filter by County"
  >
    <option
      className="filter-names"
      value="all-counties"
    >
      All Counties
    </option>
    <option
      className="filter-names"
      value="San Luis Obispo County"
    >
      San Luis Obispo County
    </option>
  </select>
</div>
`;
